src/snowflake/snowpark/functions.py

```diff
@@ -12067,54 +12059,244 @@ def fl_is_image(e: ColumnOrName, _emit_ast: bool = True) -> Column:
+            where the number is at least 0 and less than the number of expressions specified.
+            The first expression is substituted for {0}, the second for {1}, and so on.
+        *exprs: Expressions whose values will eventually be substituted into the template string
+            in place of the numbered placeholders. These can be column names or other expressions.
+            Values can be of any type coercible to a string (for example, VARCHAR, NUMBER, etc.), or FILE.
+
+    Returns:
+        A SQL OBJECT with the following structure:
+        ``{ 'template': '<template_string>', 'args': ARRAY(<value_1>, <value_2>, ...) }``
+
+        The args array contains the value of the expressions specified in the PROMPT function call.
+
+    Note:
+
+        - This function does not perform any string formatting itself.
+          It is intended to construct an object to be consumed by Cortex AI functions.
+
+        - It is an error to use a placeholder in the template string that does not have a corresponding expression,
+          but it is not an error to have expressions that are not used in the template string.
+
+    Examples::
+
+        >>> df = session.create_dataframe([["Alice", "Monday"], ["Bob", "Tuesday"]], schema=["name", "day"])
+        >>> df.select(prompt("Hello, {0}. Today is {1}", col("name"), col("day")).alias("greeting")).show()
+        --------------------------------------------
+        |"GREETING"                                |
+        --------------------------------------------
+        |{                                         |
+        |  "args": [                               |
+        |    "Alice",                              |
+        |    "Monday"                              |
+        |  ],                                      |
+        |  "template": "Hello, {0}. Today is {1}"  |
+        |}                                         |
+        |{                                         |
+        |  "args": [                               |
+        |    "Bob",                                |
+        |    "Tuesday"                             |
+        |  ],                                      |
+        |  "template": "Hello, {0}. Today is {1}"  |
+        |}                                         |
+        --------------------------------------------
+        <BLANKLINE>
+    """
+    function_name = "prompt"
+    ast = (
+        build_function_expr(function_name, [template_string] + [expr for expr in exprs])
+        if _emit_ast
+        else None
+    )
+    exprs = [_to_col_if_str(expr, function_name) for expr in exprs]
+    return _call_function(
+        function_name, lit(template_string), *exprs, _ast=ast, _emit_ast=_emit_ast
+    )
+
+
 @private_preview(version="1.29.0")
 @publicapi
 def ai_filter(
-    predicate: ColumnOrLiteralStr, expr: ColumnOrLiteralStr, _emit_ast: bool = True
+    predicate: ColumnOrLiteralStr,
+    file: Optional[ColumnOrLiteralStr] = None,
```

Comment:
can we call out the signature update in changelog

Reply:
done